frontend/lib/types/CompositeType.cpp

```diff
@@ -96,21 +96,38 @@ void CompositeType::stringify(std::ostream& ss,
 
   auto sorted = sortedSubstitutions();
 
-  if (superType || !sorted.empty()) {
+  bool printSupertype =
+    superType != nullptr && stringKind != StringifyKind::CHPL_SYNTAX;
+
+  if (printSupertype || !sorted.empty()) {
```

Comment:
I don't think this is quite equivalent to just checking for ``superType != nullptr``. Can you add a case that prints under CHPL_SYNTAX, even if it's only something like ``<TODO: classname>``?

Reply:
Sorry, I'm not sure I'm following your question.

My understanding of this logic is that it prints {base class if needed, substitutions if type is generic}. If it prints _anything_ while doing that, it also needs to include parentheses; otherwise, there's no need for parentheses. 

Second, I don't think Chapel (the language) prints the supertype of a class when printing the class. Thus, `superType` should not be printed at all, even if it exists, if `CHPL_SYNTAX` is on.

Putting these together, it seems like `printSupertype` is appropriate here; the condition for printing parentheses is on whether something will be printed at all. The supertype will only be printed if CHPL_SYNTAX is not set, and that's the only situation (modulo substitutions) where parenths will be needed.